packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,365 @@
+   */
+  required?: boolean;
+  /**
+   * Override or extend the styles applied to the component
+   */
+  style?: React.CSSProperties;
+};
+
+const formatBytes = (bytes: number): string => {
+  if (bytes === 0) {
+    return '0 bytes';
+  }
+
+  const sizeUnits: string[] = ['B', 'kB', 'MB', 'GB', 'TB'];
+  const sizeUnitIndex = Math.floor(Math.log(bytes) / Math.log(1024));
+  const sizeInUnit = bytes / 1024 ** sizeUnitIndex;
+  return `${sizeUnitIndex < 2 ? Math.round(sizeInUnit) : sizeInUnit.toFixed(1)} ${sizeUnits[sizeUnitIndex]}`;
+};
+
+const getRemoveButtonLabel = (language: Language): string => {
+  return {
+    en: 'Remove',
+    fi: 'Poista',
+    sv: 'Ta bort',
+  }[language];
+};
+
+const getRemoveButtonAriaLabel = (language: Language, fileName: string): string => {
+  return {
+    en: `Remove ${fileName} from the added files list.`,
+    fi: `Poista tiedosto ${fileName} lisättyjen tiedostojen listasta.`,
+    sv: `Ta bort ${fileName} tillagda filen från fillistan.`,
+  }[language];
+};
+
+const getFileListAriaLabel = (language: Language): string => {
+  return {
+    en: 'Added files',
+    fi: 'Lisätyt tiedostot',
+    sv: 'Tillagda filerna',
+  }[language];
+};
+
+const getRemoveSuccessMessage = (language: Language): string => {
+  return {
+    en: 'File removed.',
+    fi: 'Tiedosto poistettu.',
+    sv: 'Filen borttagen.',
+  }[language];
+};
+
+const getAddSuccessMessage = (language: Language): string => {
+  return {
+    en: 'File added.',
+    fi: 'Tiedosto lisätty.',
+    sv: 'Filen har blivit tillagd.',
+  }[language];
+};
+
+export const FileInput = ({
+  id,
+  label,
+  buttonLabel,
+  language = 'en',
+  disabled,
+  dragAndDrop,
+  className = '',
+  errorText,
+  helperText,
+  onChange,
+  required,
+  style,
+  accept,
+  multiple,
+}: FileInputProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [selectedFiles, setSelectedFiles] = useState<File[]>([]);
+  const [successText, setSuccessText] = useState<string | undefined>();
+  const hasFilesSelected = selectedFiles && selectedFiles.length > 0;
+  const fileListId = `${id}-list`;
+  const fileListRef = useRef<HTMLUListElement>(null);
+  const fileListFocusIndexRef = useRef<number>();
+  const hasDragAndDrop = !!dragAndDrop;
+  const dropAreaRef = useRef<HTMLDivElement>(null);
+  const [isDragOverDrop, setIsDragOverDrop] = useState<boolean>(false);
+
+  const wrapperProps = {
+    className,
+    helperText,
+    successText,
+    errorText,
+    id,
+    label,
+    required,
+    style,
+  };
+
+  const passButtonClickToInput = () => {
+    if (inputRef.current) {
+      inputRef.current.click();
+    }
+  };
+
+  const resetFileInputValue = () => {
+    if (inputRef.current) {
+      inputRef.current.value = '';
+    }
+  };
+
+  const handleSingleFileChange = (files: File[]) => {
+    if (files.length > 0) {
+      setSelectedFiles(files);
+      setSuccessText(getAddSuccessMessage(language));
+    } else {
+      setSuccessText(undefined);
```

Comment:
Should there be `setSelectedFiles([])` here? If user removes the file, would we want to set the selectedFiles empty, so that the onChange callback fires?

Reply:
When the user clicks "cancel" in the file window, it triggers onChange with an empty list. I'm not sure if we want to remove the current file then 🤔

Reply:
Oh good point! We would somehow need to know if it was an actual removal, or just cancelled operation, in order to trigger onChange correctly